core/m3u8lib/parser.py

```diff
@@ -14,32 +15,34 @@ def fetch_playlist_links(session: requests.Session, playlist_url: str
         The session via which the request will be made
     playlist_url : str
         The url where the m3u8 playlist is hosted
+    keep : bool
+        Keep file with links on disk.
 
     Returns
     -------
     List[str]
         A list with all the urls where the .ts files are hosted
     """
-    res: requests.Response = session.get(playlist_url, timeout=60)
-
-    with open("links.txt", "wb") as file:
-        file.write(res.content)
-
-    with open("links.txt") as file:
-        temp = file.readlines()
-        temp = [link.strip() for link in temp]
-
-    if not keep:
-        os.unlink("links.txt")
-
-    parsed_url = urlparse(playlist_url)
-    base_url: str = f"{parsed_url.scheme}://{parsed_url.netloc}{'/'.join(parsed_url.path.split('/')[:-1])}/"
-
-    # Construct a links list containing the links joined with the base_url.
-    # If the link in the m3u8 playlist is already a full link we add that to the result `links list`
-    # else we join the base_url with the link partial in the playlist and add that to the `links list`
-    links: List[str] = [(link if "https" in link else urljoin(base_url, link))
-                        for link in temp if "EXT" not in link]
+    filename = "links.txt"
+    p = pathlib.Path(playlist_url)
+    if p.is_file():
+        with open(filename) as file:
+            temp = file.readlines()
+        links: List[str] = [link.strip() for link in temp]
```

Comment:
A m3u8 file will look somewhat like this:-

 _#EXTM3U
 #EXTINF:419,Alice in Chains - Rotten Apple
 Alice in Chains_Jar of Flies_01_Rotten Apple.mp3
 #EXTINF:260,Alice in Chains - Nutshell
 Alice in Chains_Jar of Flies_02_Nutshell.mp3
 #EXTINF:255,Alice in Chains - I Stay Away
 Alice in Chains_Jar of Flies_03_I Stay Away.mp3
 #EXTINF:256,Alice in Chains - No Excuses
 Alice in Chains_Jar of Flies_04_No Excuses.mp3
 #EXTINF:157,Alice in Chains - Whale And Wasp
 Alice in Chains_Jar of Flies_05_Whale And Wasp.mp3
 #EXTINF:263,Alice in Chains - Don't Follow
 Alice in Chains_Jar of Flies_06_Don't Follow.mp3
 #EXTINF:245,Alice in Chains - Swing On This
 Alice in Chains_Jar of Flies_07_Swing On This.mp3_

if you just strip the "link" and add it to the "links" list, it will contain uneeded lines like #EXTINF , etc.
if you see the function documentation description, it should return a list of valid links, that can be donwloaded.

Reply:
I changed logic to skip any line started with `#`